api/scpca_portal/test/test_readme_file.py

```diff
@@ -0,0 +1,128 @@
+from django.test import TestCase
+
+from scpca_portal import common, readme_file, utils
+from scpca_portal.test.factories import ProjectFactory
+
+README_DIR = common.EXPECTED_VALUES_PATH / "readmes"
+
+
+class TestReadmeFileContents(TestCase):
+    def assertReadmeContents(self, expected_file_path: str, result_content: str) -> None:
+        def get_updated_content(content: str) -> str:
+            """
+            Replace the placeholder TEST_TODAYS_DATE in test/expected_values/readmes
+            with today's date for format testing.
+            """
+            content = content.replace(
+                "Generated on: TEST_TODAYS_DATE", f"Generated on: {utils.get_today_string()}"
+            )
+
+            return content.strip()
+
+        with open(expected_file_path, encoding="utf-8") as expected_file:
+            expected_content = get_updated_content(expected_file.read())
+        # Convert expected and result contents to line lists for easier debugging
+        self.assertEqual(
+            expected_content.splitlines(True),
+            result_content.splitlines(True),
+            f"{self._testMethodName}: Comparison with {expected_file_path} does not match.",
+        )
+
+    def test_readme_file_PORTAL_ALL_METADATA(self):
+        DOWNLOAD_CONFIG_NAME = "PORTAL_ALL_METADATA"
+        DOWNLOAD_CONFIG = common.PORTAL_METADATA_DOWNLOAD_CONFIG
+        PROJECT_IDS = ["PROJECT_ID_0", "PROJECT_ID_1", "PROJECT_ID_2"]
+
+        projects = [
+            ProjectFactory(
+                additional_restrictions="Research or academic purposes only", scpca_id=project_id
```

Comment:
Instead of passing this everywhere you should try adding it as a default for the Project factory and remove it from here (and elsewhere in this file). I think that is a sensible factory default.